### DISCUSSION TITLE
Use semantic versioning for releases

### DISCUSSION THREAD

**Discussion by hfehrmann** [immich-app/immich#11367] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Since there is no stable release yet, and sometimes there are breaking changes between releases. Is it possible to use semantic versioning?

You folks already do it for hotfixes, why not include the breaking changes into the bake?

This would help to identify how to run migrations for each breaking change, with immediate feedback without lurking all the releases

I found discussion #2636, but the argument of stability doesn't make sense IMHO, or there could be a 4 number versioning if you want to have a v2.0.0.0 to be the real stable

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
Immich is currently under heavy development, and unfortunately the expectation has become (even before we set that explicitly) that 2.0.0 will be a stable release. In terms of semver, you should think of Immich currently as being a 0.x.x version.

**Reply by hfehrmann** [reply]
You could still have 0.x.y.z where x increase for breaking changes, y for incremental improvement and z for hotfix.

What is going to happen when Immich reach stable.
- Is there never going to be a new breaking change? 
- Would every new breaking change be a new version?

**Reply by bo0tzz** [reply]
When Immich reaches stable we will start adhering to semver (ie breaking changes will be major versions) and we will make much fewer breaking changes.
